src/user/tidal_bay_initialization.F90

```diff
@@ -44,6 +47,10 @@ function register_tidal_bay_OBC(param_file, CS, OBC_Reg)
   endif
   allocate(CS)
 
+  call get_param(param_file, mdl, "TIDAL_BAY_FLOW", CS%tide_flow, &
+                 "Maximum total tidal volume flux.", &
+                 units="m3 s-1", default=3.0d6, scale=US%m_s_to_L_T*US%m_to_L*US%m_to_Z)
```

Comment:
With the gnu/10.1.0 compiler, we get a compilation error due to dummy arg specification `default=3.0d6`. Changing that to `default=3.0e6` fixes the issue for us.

```
   52 |                  units="m3 s-1", default=3.0d6, scale=US%m_s_to_L_T*US%m_to_L*US%m_to_Z)
      |                                                                                        1
Error: There is no specific subroutine for the generic ‘get_param’ at (1)
```

Reply:
I wonder if this is a compiler bug, maybe related to our kind-less `real` declarations and `-r8` flag usage.  Do you see this in 10.2?  I see no issues with 10.2 and 11.1.  Just asking out of interest, no need to investigate further.

Either way, easy enough to fix.

Reply:
My bad. Adding the `-fdefault-double-8` compiler flag fixes the issue. It's interesting that this dummy arg is the only instance that requires the `-fdefault-double-8` flag, but that's fine.

Reply:
Personally I do not see any reason why we should not change the dummy argument specification to `default=3.0e6`, as suggested by @alperaltuntas, to correct the problem without requiring any new compiler flags and to match the pattern used elsewhere in the MOM6 code.  This seems to be a case where correcting the problem twice might be in order.